### PR TITLE
feat(model): add model diff api for server

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description

1. tune the response's structure of model info api from flatten list to node list
2. add model diff api
3. refactor model download api

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

**Comment by dreamlandliu** [comment]
lgtm

**Review by dreamlandliu — COMMENTED** [review]
lgtm

File: server/controller/src/main/java/ai/starwhale/mlops/domain/model/ModelService.java [hunk 1]
```diff
@@ -266,6 +292,23 @@ private ModelInfoVo toModelInfoVo(ModelEntity model,
...
+                var fork = FileNode.parse(fileDesc.getPath(), (node) -> {
+                    node.setSize(FileUtil.readableFileSize(fileDesc.getSize()));
+                    node.setDesc(fileDesc.getDesc().name());
+                    node.setSignature(fileDesc.getSignature());
+                });
+                // merge to files
+                FileNode.merge(files, fork);
+            }
+        }
+        return files;
```
Comment (anda-ren) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/model/ModelService.java:309]: could we combine `parse` and `merge` to one method like `makeTree` which tasks `meta.getResources()` as a param.
Such then:
1. return `FileNode.makeTree` here
2. reuse `FileNode.makeTree`  when runtime diff or dataset diff

Comment (goldenxinxing) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/model/ModelService.java]: done

**Review by anda-ren — APPROVED** [review]
lgtm

**Pull request merged**